Tasks/SonarQubePreBuild/SonarQubePreBuild.ps1

```diff
@@ -21,6 +21,17 @@ Write-Verbose -Verbose "configFile = $configFile"
 Write-Verbose -Verbose "dbConnectionString = $dbUrl"
 Write-Verbose -Verbose "dbUsername = $dbUsername"
 
+$prcaEnabled = Get-TaskVariable -Context $distributedTaskContext -Name "GitPullRequestCodeAnalysisEnabled"
+if ($prcaEnabled -ieq "true")
+{
+    Write-Verbose -Verbose "GitPullRequestCodeAnalysisEnabled is true, setting cmdline args for incremental mode for sonar-runner..."
+
+    if (!$cmdLineArgs.ToString().Contains("sonar.analysis.mode"))
```

Comment:
If sonar.analysis.mode is set to something else already, then the execution won't go inside the ''if' condition and we won't do anything. I'll change the logic a bit in the post-test task to handle this scenario..